XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs

```diff
@@ -0,0 +1,159 @@
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using XenAdmin.Actions;
+using XenAdmin.Core;
+using XenAdmin.Dialogs.ServerUpdates;
+using XenAdmin.Network;
+using XenAPI;
+
+
+namespace XenAdmin.Alerts
+{
+    public class OutOfSyncWithCdnAlert : Alert
+    {
+        private readonly int _outOfSyncDays;
+        private readonly Pool _pool;
+ 
+        private OutOfSyncWithCdnAlert(Pool pool, DateTime timestamp)
+        {
+            _timestamp = timestamp;
+            _pool = pool;
+            Connection = _pool.Connection;
+
+            if (_timestamp - _pool.last_update_sync >= TimeSpan.FromDays(180))
+            {
+                _outOfSyncDays = 180;
+                Priority = AlertPriority.Priority1;
+            }
+            else if (_timestamp - _pool.last_update_sync >= TimeSpan.FromDays(90))
+            {
+                _outOfSyncDays = 180;
+                Priority = AlertPriority.Priority2;
+            }
+        }
+
+        public static bool TryCreate(IXenConnection connection, out Alert alert)
+        {
+            if (Helpers.XapiEqualOrGreater_23_18_0(connection))
+            {
+                var pool = Helpers.GetPoolOfOne(connection);
+                var timestamp = DateTime.UtcNow;
+
+                if (timestamp - pool.last_update_sync >= TimeSpan.FromDays(90))
+                {
+                    alert = new OutOfSyncWithCdnAlert(pool, timestamp);
+                    return true;
+                }
+            }
+
+            alert = null;
+            return false;
+        }
+
+        public override AlertPriority Priority { get; }
+
+        public override string AppliesTo => Helpers.GetName(_pool);
+
+        public override string Description => Title;
+
+        public override Action FixLinkAction
+        {
+            get
+            {
+                return () =>
+                {
+                    var syncAction = new SyncWithCdnAction(_pool);
+                    syncAction.Completed += a => Updates.CheckForCdnUpdates(a.Connection);
+                    syncAction.RunAsync();
+                };
+            }
+        }
+
+        public override string FixLinkText => Messages.UPDATES_GENERAL_TAB_SYNC_NOW;
+
+        public override string HelpID => "TODO";

```

Comment:
TODO value, shouldnt this have a value now?

Reply:
No help IDs are known yet.